templates/tests/server-test.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.server.enabled }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ .Release.Name }}-server-test"
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: {{ .Release.Name }}-server-test
+      image: {{ .Values.server.image.repository }}:{{ .Values.server.image.tag | default "latest" }}
```

Comment:
I'm surprised to see the `| default "latest"` snippet here. Given we package the chart with a default tag in `values.yaml`, is it possible for tag to not be set? Unless people really go out of their way I suppose..

Reply:
Yeah, good catch, removed it in a498b29.

Reply:
Well I take it back. It looks like since the test is templated as part of the chart install, there's a test that fails if the tag doesn't default to "latest" like this: https://github.com/hashicorp/vault-helm/blob/ec67b5dd45c60346f709630e0d9acbd6f538ac69/test/unit/server-dev-statefulset.bats#L49-L60

So put it back for now.